### PR TITLE
Allow slack channels to be specified

### PR DESCRIPTION
**Pull request by sidravi1**
Hi @yuhongsun96 and @Weves ,

Here's a go at allowing channels to be specified for the slack connector. Feedback most welcome!

image

I didn't have an easy/non-invasive way to test that not specifying channels still works. Is this something you can easily test?

### CODE REVIEW

File: web/src/app/admin/connectors/slack/page.tsx [hunk 1]
```diff
@@ -170,21 +179,36 @@ const MainSection = () => {
       <div className="border-solid border-gray-600 border rounded-md p-6 mt-4">
         <h2 className="font-bold mb-3">Connect to a New Workspace</h2>
         <ConnectorForm<SlackConfig>
-          nameBuilder={(values) => `SlackConnector-${values.workspace}`}
+          nameBuilder={(values) =>
+            `SlackConnector-${values.workspace}-${values.channels.join("_")}`
```
Comment (sidravi1) [inline web/src/app/admin/connectors/slack/page.tsx]: This name could get long! Is it a concern?

Comment (Weves) [reply web/src/app/admin/connectors/slack/page.tsx]: This should be fine! That column does not have a specified maximum varchar length, so it should be able to be very, very long without any issues.

File: web/src/app/admin/connectors/slack/page.tsx [hunk 2]
```diff
@@ -152,6 +155,12 @@ const MainSection = () => {
                   getValue: (connector) =>
                     connector.connector_specific_config.workspace,
                 },
+                {
+                  header: "Channels",
+                  key: "channels",
+                  getValue: (connector) =>
+                    connector.connector_specific_config.channels,
```
Comment (Weves) [inline web/src/app/admin/connectors/slack/page.tsx]: `getValue` is "supposed" to (according to the typing) return a `string` or `JSX.Element` not `string[]`. Can we do something like  `connector.connector_specific_config.channels.join(", ")` to both address this typing issue + to make it display better in the UI?

File: web/src/lib/types.ts [hunk 3]
```diff
@@ -65,6 +65,7 @@ export interface ProductboardConfig {}
 
 export interface SlackConfig {
   workspace: string;
+  channels: string[];
```
Comment (Weves) [inline web/src/lib/types.ts]: Can we make this change backwards compatible by marking it as optional (e.g. `channels?: string[]`)? This also involves handling this `undefined` channels case in a few other places as well.

File: web/src/components/admin/connectors/ConnectorForm.tsx [hunk 4]
```diff
@@ -119,7 +119,8 @@ export function ConnectorForm<T extends Yup.AnyObject>({
       >
         {({ isSubmitting, values }) => (
           <Form>
-            {formBody ? formBody : formBodyBuilder && formBodyBuilder(values)}
+            {formBody && formBody}
```
Comment (Weves) [inline web/src/components/admin/connectors/ConnectorForm.tsx:130]: nit: can we change the comment on line 45 to account for this change?

**Comment by Weves** [comment]
Overall looks great! I tested the case of not specifying any channels and works as expected. 

Added a few small comments! If you'd prefer, I'm also happy to address them myself.

**Comment by sidravi1** [comment]
Hey @Weves - I don't see any comments? Happy to make any fixes needed.

**Comment by Weves** [comment]
@sidravi1 oops, forgot to submit the review. Comments should be visible now. Thanks!

File: web/src/lib/types.ts (hunk 3 above)
Comment (sidravi1) [reply web/src/lib/types.ts]: ok! I think i also got the other places where we need to handle `undefined`.

**Comment by sidravi1** [comment]
@Weves - Just checking that you saw that this was ready for your review!

**Review by Weves — APPROVED** [review]
Sorry about the delay - looks good to me! Thanks a bunch for adding this 🙇

**Pull request merged**